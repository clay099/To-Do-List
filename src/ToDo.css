.completed {
	text-decoration: line-through;
}

@keyframes fadeIn {
	0% {
		opacity: 0;
	}
	100% {
		opacity: 1;
	}
}

.ToDo {
	margin: 5px 5px;
	animation: fadeIn ease-in 1s;
}

.ToDo button {
	margin-left: 5px;
}

.ToDo-delete {
	background-color: rgb(255, 138, 138);
}

.ToDo-edit {
	background-color: lightgreen;
}

.ToDo-complete {
	background-color: rgb(68, 202, 255);
}
