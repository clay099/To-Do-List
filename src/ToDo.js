import React, { useState } from "react";
import ToDoList from "./ToDoList";
import "./ToDo.css";

const ToDo = ({ task, deleteToDo, editTodo }) => {
	const [editTask, setEditTask] = useState(task);
	const [isEditing, setIsEditing] = useState(false);
	const [isComplete, setIsComplete] = useState(false);

	const handleChange = (e) => {
		setEditTask(e.target.value);
	};

	const toggleEdit = () => {
		setIsEditing((edit) => !edit);
	};

	const handleUpdate = (e) => {
		e.preventDefault();
		editTodo(task, editTask);
		setIsEditing(false);
	};

	const toggleComplete = () => {
		setIsComplete((val) => !val);
	};

	const completeClass = isComplete ? " ToDo completed" : "ToDo";

	let display = (
		<div className={completeClass} onClick={toggleComplete}>
			{task}
			<button
				className="ToDo-delete"
				onClick={() => {
					deleteToDo(task);
				}}
			>
				Delete
			</button>
			<button className="ToDo-edit" onClick={toggleEdit}>
				Edit
			</button>
			<button className="ToDo-complete">Mark Complete</button>
		</div>
	);

	if (isEditing) {
		display = (
			<div className="ToDo">
				<form onSubmit={handleUpdate}>
					<input type="text" value={editTask} onChange={handleChange} />
					<button className="ToDo-update">Update!</button>
				</form>
			</div>
		);
	}

	return display;
};

export default ToDo;
