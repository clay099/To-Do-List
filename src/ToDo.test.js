import React from "react";
import { render } from "@testing-library/react";
import ToDo from "./ToDo";

//smoke test
it("should render NewBoxForm", () => {
	render(<ToDo />);
});

//snapshot test
it("should match snapshot", () => {
	const { asFragment } = render(<ToDo />);
	expect(asFragment()).toMatchSnapshot();
});
