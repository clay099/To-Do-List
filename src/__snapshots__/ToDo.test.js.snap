// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDo"
  >
    <button
      class="ToDo-delete"
    >
      Delete
    </button>
    <button
      class="ToDo-edit"
    >
      Edit
    </button>
    <button
      class="ToDo-complete"
    >
      Mark Complete
    </button>
  </div>
</DocumentFragment>
`;
